java/README.md

```diff
@@ -1,311 +1,98 @@
-to the other language mappings. The script uses Ice for Python to communicate
-with the Android app, therefore you must build the [Python mapping]
-(../python) before continuing.
-
-You also need to add the `tools\bin`, `platform-tools` and `emulator`
-directories from the Android SDK to your PATH. On macOS, you can use the
-following commands:
-
-```
-export PATH=~/Library/Android/sdk/cmdline-tools/latest/bin:$PATH
-export PATH=~/Library/Android/sdk/platform-tools:$PATH
-export PATH=~/Library/Android/sdk/emulator:$PATH
-```
-
-On Windows, you can use the following commands:
-
-```
-set PATH=%LOCALAPPDATA%\Android\sdk\cmdline-tools\latest\bin;%PATH%
-set PATH=%LOCALAPPDATA%\Android\sdk\platform-tools;%PATH%
-set PATH=%LOCALAPPDATA%\Android\sdk\emulator;%PATH%
-```
-
-Run the tests with the Android emulator by running the following command:
-
-```
-python allTests.py --android --controller-app
-```
-
-To run the tests on a Android device connected through USB, you can use
-the `--device=usb` option as shown below:
-
-```
-python allTests.py --android --device=usb --controller-app
-```
-
-To connect to an Android device that is running adb you can use the
-`--device=<ip-address>`
-
-```
-python allTests.py --android --device=<ip-address> --controller-app
-```
-
-To run the tests against a `controller` application started from Android
-Studio you should omit the `--controller-app` option from the commands above.
-
-## IceGrid GUI Tool
-
-Ice for Java includes the IceGrid GUI tool. It can be found in the file
-`lib/icegridgui.jar`.
-
-This JAR file is completely self-contained and has no external dependencies.
-You can start the tool with the following command:
-```
-java -jar icegridgui.jar
-```
-
-On macOS, the build also creates an application bundle named IceGrid GUI. You
-can start the IceGrid GUI tool by double-clicking the IceGrid GUI icon in
-Finder.
-
-[1]: https://zeroc.com/downloads/ice
-[2]: https://doc.zeroc.com/ice/3.7/release-notes/supported-platforms-for-ice-3-7-100
-[3]: https://gradle.org
-[4]: http://proguard.sourceforge.net
-[5]: https://commons.apache.org/proper/commons-compress/
+# Ice for Java
+
+[Getting started] | [Examples] | [Maven packages] | [Documentation] | [Building from source]
+
+The [Ice framework] provides everything you need to build networked applications, including RPC, pub/sub, server deployment, and more.
+
+Ice for Java is the Ice Java implementation.
+
+## Sample Code
+
+```slice
+// Slice definitions (Hello.ice)
+
+#pragma once
+
+["java:package:com.zeroc.demos.Ice.minimal"]
+module Demo
+{
+    interface Hello
+    {
+        void sayHello();
+    }
+}
+```
+
+```java
+// Client implementation (Client.java)
+
+import com.zeroc.demos.Ice.minimal.Demo.*;
+
+public class Client
+{
+    public static void main(String[] args)
+    {
+        try(com.zeroc.Ice.Communicator communicator = com.zeroc.Ice.Util.initialize(args))
+        {
+            communicator.getProperties().setProperty(
```

Comment:
Why do we do that?

Reply:
We have this in the minimal demo, but we can get rid of it as this demo doesn't use classes or exceptions.

Reply:
Removed